ZIO 2.0.0-RC1 upgrade

yisraelU opened the pull request:
upgraded to RC1 of ZIO 2.0.0
Added AMQPConfig class with default config and new connection method

svroonland commented:
Nice!

I made a fix for the failing build in this commit, perhaps you could cherry-pick it:  `c57c271883c277632121a508f380f47878cbfc12`. I can't push to your master branch, but I believe if you make a PR from a branch on your fork, I can push changes to it as well.

svroonland commented on src/main/scala/nl/vroste/zio/amqp/model/AMQPConfig.scala:
```diff
@@ -0,0 +1,33 @@
+  user: String,
+  password: String,
+  vhost: String,
+  heartbeatInterval: Short,
```
Shall we make this a zio Duration so we know what the unit is (now I'm not sure if it's seconds or milliseconds)

yisraelU replied:
Yes , I'll do that for all time related fields

svroonland commented on src/main/scala/nl/vroste/zio/amqp/model/AMQPConfig.scala:
```diff
@@ -0,0 +1,33 @@
+  ssl: Boolean,
+  host: String,
+  port: Short,
+  connectionTimeout: Short
```
Shall we make this a zio Duration?

svroonland commented on project/Dependencies.scala:
```diff
@@ -1,11 +1,13 @@
     "dev.zio"                %% "zio-streams"                 % zioVersion,
     "dev.zio"                %% "zio-test"                    % zioVersion % Test,
     "dev.zio"                %% "zio-test-sbt"                % zioVersion % Test,
+    "dev.zio"                %% "zio-config"                  % "1.0.10",
```
I don't think this is used

yisraelU replied:
True I was planning on using it for other code for loading config , just didn't want to push that info this commit .
I'll remove it

svroonland reviewed (changes requested):
Looking good, a few remarks

yisraelU commented:
> Nice!
> 
> I made a fix for the failing build in this commit, perhaps you could cherry-pick it: `c57c271883c277632121a508f380f47878cbfc12`. I can't push to your master branch, but I believe if you make a PR from a branch on your fork, I can push changes to it as well.
@svroonland 
Seems like unused imports cause failure. I removed them.
The test thats failing now needs to be changed as it runs the wrong version of Scala 3. which is incompatible with ZIO 2.0.0 RC1 . in the build.sbt I switched from 3.0.1 to 3.1.0 , however the tests are hardcoded to 3.0.1

svroonland commented:
Yeah for that we need to make the changes to the config.yml file like in the commit I referenced.

yisraelU commented:
> Yeah for that we need to make the changes to the config.yml file like in the commit I referenced.

@svroonland sorry , I missed that.
All done

svroonland commented:
Great, thanks! I merged it, it should be in the snapshot soon.
